rbac/default-role-manager/role_manager.go

```diff
@@ -557,6 +574,17 @@ func (dm *DomainManager) getRoleManager(domain string, store bool) *RoleManagerI
 				return true
 			})
 		}
+		if dm.tenantManager != nil {
```

Comment:
Even though I do not know much about `golang`, this feels little repetitive, don't you think?

I think some comments would also be warranted. These specific conditions should be explained, otherwise somebody would just scratch their head in the future when they try to figure out why these are done as they are.

Reply:
I don't quite understand what repetition means, what is your suggestion

Reply:
did some refactoring